File: XPile 3-17/Test/test_match.js
function test_int_match() {
    let x = 2;
    switch (x) {
        case 1:
            console.log("one");
            break;
        case 2:
            console.log("two");
            break;
        case 3:
            console.log("three");
            break;
        default:
            console.log("other");
            break;
    }
}

function test_default() {
    let y = 99;
    switch (y) {
        case 1:
            console.log("one");
            break;
        case 2:
            console.log("two");
            break;
        default:
            console.log("default hit");
            break;
    }
}

function test_match_expr() {
    let val = (3 + 2);
    switch (val) {
        case 4:
            console.log("four");
            break;
        case 5:
            console.log("five");
            break;
        case 6:
            console.log("six");
            break;
    }
}

function test_match_negative() {
    let n = (-1);
    switch (n) {
        case (-1):
            console.log("negative one");
            break;
        case 0:
            console.log("zero");
            break;
        case 1:
            console.log("positive one");
            break;
    }
}

function main() {
    test_int_match();
    test_default();
    test_match_expr();
    test_match_negative();
    return 0;
}


main();

File: XPile 3-17/Test/test_match.py
from dataclasses import dataclass, field

def test_int_match():
    x = 2
    match x:
        case 1:
            print('one')
        case 2:
            print('two')
        case 3:
            print('three')
        case _:
            print('other')

def test_default():
    y = 99
    match y:
        case 1:
            print('one')
        case 2:
            print('two')
        case _:
            print('default hit')

def test_match_expr():
    val = (3 + 2)
    match val:
        case 4:
            print('four')
        case 5:
            print('five')
        case 6:
            print('six')

def test_match_negative():
    n = (-1)
    match n:
        case (-1):
            print('negative one')
        case 0:
            print('zero')
        case 1:
            print('positive one')

def main():
    test_int_match()
    test_default()
    test_match_expr()
    test_match_negative()
    return 0


if __name__ == "__main__":
    main()
